src/vmm/src/device_manager/mmio.rs

```diff
@@ -179,7 +189,7 @@ impl MMIODeviceManager {
     ) -> Result<(), MmioError> {
         // Our virtio devices are currently hardcoded to use a single IRQ.
         // Validate that requirement.
-        if device_info.irqs.len() != 1 {
+        if device_info.irq.is_none() {
             return Err(MmioError::InvalidIrqConfig);
         }
```

Comment:
We can avoid the `unwrap()` below by doing

```suggestion
        let Some(irq) = device_info.irq else {
            return Err(MmioError::InvalidIrqConfig)
        };
```